modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala

```diff
@@ -107,8 +108,9 @@ class ProjectionMonitorActor extends Actor with ActorLogging {
 
     val monitorName = s"monitor::$projectionName::${commandId.value}::${UUID.randomUUID()}"
 
-    // TODO this could be moved to property file eventually
-    val shutdownEventsMonitorAfter = 10.seconds
+    // take at least the same timeout as the underlying projection
+    // if you're willing to wait X seconds on the projection, you should be willing to wait X seconds for events coming from the Projection
+    val shutdownEventsMonitorAfter = projectionConfig(projectionName).getDuration("async-projection-timeout", 5.seconds) + 1.second
```

Comment:
maybe instead `+ 1.second` we should have a `* factor`? What do you think?

Reply:
That would required an extra param

Reply:
Actually, thinking about it a bit more, maybe the "proper" value for this timeout should depend on the askTimeout you're using to talk to the AggregateActorRef, but that would require much more intrusive changes.

I'm willing to add the extra factor if you feel it's necessary. I don't think it's that useful to be honest.

Reply:
that's ok, let's leave it like that.